### PR TITLE
refactor(integrations): deprecate and proxy the langchain integration

### PR DESCRIPTION
**Pull request by parambharat**
Description
-----------
This PR deprecates the existing `langchain` integration in favor of the import from `langchain`.
Currently, it provides a proxy import for the `WandbTracer` from `langchain` and includes a deprecation message for the next release. Additionally, we change the telemetry to track the import label rather than the feature.

### <samp>🤖 Generated by Copilot at ce8dfe9</samp>

This pull request updates the LangChain integration with W&B to use the new langchain package and removes the old one. It also fixes the feature code for the LangChain W&B Tracer callback in the yea tests. It affects the files `wandb/integration/langchain/wandb_tracer.py`, `wandb/integration/langchain/util.py`, and two yea test files.

Testing
-------
How was this PR tested?

Ensured that the yea test run successfully.

- [X] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at ce8dfe9</samp>

> _Sing, O Muse, of the skillful code reviewers_
> _Who scrutinized the changes of the LangChain tracer_
> _And found the feature codes mismatched and erroneous_
> _And urged the swift correction of the wandb/sdk/lib/telemetry.py maker_

### CODE REVIEW

File: wandb/integration/langchain/wandb_tracer.py [hunk 1]
```diff
@@ -14,197 +14,31 @@
...
-        #     model_dict = safely_convert_model_to_dict(model)
-
-        model_trace = trace_tree.WBTraceTree(
-            root_span=root_span,
-            model_dict=model_dict,
+class WandbTracer(WandbTracer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        wandb.termwarn(
+            "This integration has been moved to langchain. In the future, please use:`from "
```
Comment (parambharat) [inline wandb/integration/langchain/wandb_tracer.py:41]: Mention that the preferred method is to set the env var.  Add link to docs.

**Review by dmitryduev — CHANGES_REQUESTED** [review]
thanks @parambharat, agreed with the general approach, please see my comments.

File: wandb/integration/langchain/wandb_tracer.py [hunk 2]
```diff
@@ -14,197 +14,33 @@
...
+
+class WandbTracer(WandbTracer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        wandb.termwarn(
+            "This integration has been moved to langchain. The preferred method to enable the tracer is to set the "
+            "environment variable `LANGCHAIN_WANDB_TRACING=true`. Please refer to the langchain documentation at "
+            "https://python.langchain.com/en/latest/integrations/agent_with_wandb_tracing.html  In future, "
+            "please use: `from langchain.callbacks.tracers import WandbTracer` in place of this import. This import "
+            "will be removed in a v0.15.5 of wandb"
```
Comment (dmitryduev) [inline wandb/integration/langchain/wandb_tracer.py:45]: Please rm the last sentence.

Comment (parambharat) [reply wandb/integration/langchain/wandb_tracer.py]: @dmitryduev, I changed the message recently based on comments from @morganmcg1 , I have also removed the last sentence as per your suggestion.

Comment (dmitryduev) [reply wandb/integration/langchain/wandb_tracer.py]: thanks. please also see my note above regarding using our deprecation framework.

File: tests/functional_tests/t0_main/langchain/t1_v1.yea [hunk 3]
```diff
@@ -24,4 +24,4 @@ assert:
   #   - model_hash
   - :op:contains:
     - :wandb:runs[0][telemetry][3]  # features
-    - 52  # LangChain W&B Tracer callback
+    - 23  # LangChain W&B Tracer callback
```
[the remark this answers is not in the document]
Comment (parambharat) [reply tests/functional_tests/t0_main/langchain/t1_v1.yea]: Hi @dmitryduev, I've addressed this by testing for the "langchain" label we log in [langchain integration](https://github.com/hwchase17/langchain/blob/e0e3ef1c57109ac5491ba744b8e4a4189931b1b5/langchain/callbacks/tracers/wandb.py#LL273C33-L273C33).

File: wandb/integration/langchain/wandb_tracer.py (hunk 2 above)
Comment (parambharat) [reply wandb/integration/langchain/wandb_tracer.py]: removed

File: wandb/integration/langchain/wandb_tracer.py (hunk 1 above)
Comment (parambharat) [reply wandb/integration/langchain/wandb_tracer.py]: Hi @dmitryduev , I've changed this to use the `deprecate` functionality as suggested.

**Review by dmitryduev — APPROVED** [review]
🙏🏻 many thanks @parambharat, LGTM!

**Pull request merged**